newdle/api.py

```diff
@@ -77,6 +82,45 @@ def require_token():
     g.user = user
 
 
+@api.route('/avatar/<payload>')
+@allow_anonymous
+def user_avatar(payload):
+    user_info = avatar_info_from_payload(payload)
+    size = request.args.get('size')
+    email_hex = hashlib.md5(user_info['email'].encode()).hexdigest()
+    # make gravatar return 404 HTTP code instead of a default image
+    query_args = {'d': '404'}
+    if size is not None:
+        query_args['s'] = size
+
+    gravatar_url = f'https://gravatar.com/avatar/{email_hex}?{url_encode(query_args)}'
+    request_headers = {
+        header: value
+        for header, value in request.headers.items()
+        if header in ['if-modified-since']
+    }
```

Comment:
```suggestion
    request_headers = {}
    if 'If-Modified-Since' in request.headers:
        request_headers['If-Modified-Since'] = request.headers['If-Modified-Since']
```

Reply:
maybe `request_headers = {'If-Modified-Since': request.headers.get('If-Modified-Since')}` would work as well, not sure how requests handles a `None` there...